main.rb

```diff
@@ -0,0 +1,108 @@
+require_relative 'bank_account'
+require_relative 'bank_system'
+
+# Main program
+bank_system = BankSystem.new
+
+puts "Welcome! \n What do you want to do?"
+
+loop do
+  puts "\n Press: \n 1 - create account; \n 2 - display all the accounts; \n 3 - retrieve money; \n 4 - deposit money; \n 5 - display current balance; \n 6 - exit. "
+
+  choice = gets.chomp.to_i
+
+  case choice
```

Comment:
This method is veery long and hard to follow. I would try to extract the logic into methods like this:
```ruby
case choice
when 1
  create_account
  break
when 2
  bank_system.display_accounts
  break
# .. and so on
```